### PR TITLE
perf(github): add knob/flag to manually override CI skip

### LINKED ISSUE #2679: perf(github): add knob/flag to manually override CI skip
### Description

As a maintainer I want to be able to make lots of small changes to scripts and so on so that the quality of the code is improving little bit little over time (or at least we keep it level, fighting code rot)

We have an initiative to dynamically analyze the diff so that test are only ran for packages that have code changes in them but this is hard to implement to cover edge cases like 
1. "Oh I just did a linter fix of typings (Typescript) and did not modify runtime code (JS)"
2. I just added a newline to this bash script that the linter was telling me to fix
3. I fixed an obvious bug in some build script that was failing it but I have a very high degree of confidence that this did not modify runtime behavior of production code.
4. etc.

- https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/
- https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs

### Acceptance Criteria
1. Probably the best is to make it based on commit message body (the subject line is already crowded and we don't need need development specific noise in the release notes)
2. Ideally this would be locked down to be used only by maintainers because otherwise malicious actors might try to sneak in malware by skipping security related CI checks that we have in place to prevent exactly that. I'm not sure how feasible this is but part of the issue scope here is to research the options to make that happen and write up a little report about each of those and which one could be the best and why.

### PR DESCRIPTION
**Pull request by zondervancalvez**
### **Commit** to be reviewed
---
perf(github): add knob/flag to manually override CI skip
```
Primary Changes
----------------
1. Created a function to check if the commit message
contains 'cacti-skip-ci' to skip the CI check.
2. Check if the request is from the maintainer or not.
If he is a Maintainer, the CI-skip request will be
granted, else not.
```
Fixes #2679

**Pull Request Requirements**
- [x] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [x] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [x] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [x] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [x] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

File: tools/should-skip-ci.js
Comment (jagpreetsinghsasan) [inline tools/should-skip-ci.js:68]: Can you change this to `The author of this PR is an active maintainer`?

File: tools/should-skip-ci.js
Comment (jagpreetsinghsasan) [inline tools/should-skip-ci.js:71]: Can you change this to `The author of this PR is not an active maintainer. \n Please refer to https://github.com/hyperledger/cacti/blob/main/MAINTAINERS.md for the list of active maintainers.`?

File: tools/should-skip-ci.js
Comment (jagpreetsinghsasan) [inline tools/should-skip-ci.js:17]: Please change these references to PR's belonging to Cacti repo (not `zondervan` user repo) for ease of use? This is needed so that if one needs to quickly debug any PR which fails to do so, can just change the PR number at the end of this URL and test it out, instead of changing the origin name as well from `zondervan` to `hyperledger`

File: tools/should-skip-ci.js
Comment (jagpreetsinghsasan) [inline tools/should-skip-ci.js:3]: Please add a in-code comment prior to this line specifying that this is the new tag which can be used to facilitate ci-skip

File: tools/should-skip-ci.js
Comment (jagpreetsinghsasan) [inline tools/should-skip-ci.js:84]: Please change the code comment to something like `Exit with an error code so as to pause the dependent workflows`

File: tools/should-skip-ci.js
Comment (jagpreetsinghsasan) [inline tools/should-skip-ci.js:86]: Please remove these log messages as you have already added them in `checkCommitterIsMaintainer( )` function and instead append this `CI will be skipped` to the respective log message within the above mentioned function.

File: tools/should-skip-ci.js
Comment (jagpreetsinghsasan) [inline tools/should-skip-ci.js:6]: This regex is too big for us to maintain in long run (Please refer to how we have added multiple regex in pr-commit-parity.js script). As the next steps, we can break this up into several simpler regex, which can be maintained in a better way.
Summary: Instead of doing maintainersData.replace(ONE_BIG_LONG_REGEX), you can cascade simpler, shorter regex for better maintainability like string.replace(HEADER_REGEX).replace(HYPHEN_REGEX).replace(DELIMITER_REGEX)... and so on...
I will create another issue to fix that once this gets merged

File: tools/should-skip-ci.js
Comment (jagpreetsinghsasan) [reply tools/should-skip-ci.js:4]: @zondervancalvez please update the comment to also include that it is specifically for maintainers. Also rename this file to ci-skip-for-maintainers.js

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM, just update it to the latest `main`

File: .github/workflows/ci.yaml
Comment (petermetz) [inline .github/workflows/ci.yaml:2654]: @zondervancalvez Please clean up white space changes from the diff

**Review by petermetz — COMMENTED** [review]
LGTM with comments

**Review by outSH — APPROVED** [review]

**Pull request merged**
